mixnet/protocol/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+use std::error::Error;
+
+use tokio::io::{AsyncWrite, AsyncWriteExt};
+
+pub enum BodyType {
+    SphinxPacket,
+    FinalPayload,
+}
+
+impl BodyType {
+    pub fn as_u8(&self) -> u8 {
+        match self {
+            Self::SphinxPacket => 0,
+            Self::FinalPayload => 1,
+        }
+    }
+
+    pub fn from_u8(value: u8) -> Self {
+        match value {
+            0 => Self::SphinxPacket,
+            1 => Self::FinalPayload,
+            _ => todo!("return error"),
+        }
+    }
+}
+
+pub async fn write_body<'a, W>(
+    writer: &'a mut W,
+    body_type: BodyType,
+    body: &[u8],
+) -> Result<(), Box<dyn Error>>
+where
+    W: AsyncWrite + Unpin + ?Sized,
+{
+    writer.write_u8(body_type.as_u8()).await?;
+    writer.write_all(body).await?;
+    Ok(())
+}
```

Comment:
I think that instead of having a `BodyType` type. It would be cleaner with a `Body` in this fashion:

```rust
pub enum Body {
    SphinxPacket(Vec<u8>)
    FinalPayload(Vec<u8>)
}
```
that is an owned version, can be done with a `&'a [u8]` as well inside or even using `Cow<'a>` inside for handling both owned and shared versions.
If you feel confused I can make a commit with just this changes.

Reply:
Resolved by https://github.com/logos-co/nomos-node/pull/308. Thank you.